docs/advanced/UsageWithTypescript.md

```diff
@@ -276,6 +276,48 @@ interface AppProps {
 
 With these additions made props that come from redux's side are now being type checked. Feel free to extend the interface as necessary to account for additional props being passed down from parent components.
 
+## Usage with Redux Thunk
+
+Redux Thunk is a commonly used middleware for asynchronous orchestration. Feel free to check out its documentation [here](https://github.com/reduxjs/redux-thunk). A thunk is a function that returns another function that takes parameters `dispatch` and `getState`. We will define a type which describes this in a new file:
+
+```ts
+// src/types.ts
+
+import { Dispatch } from 'redux'
+import { AppState } from './store'
+
+export type ThunkType<T> = (dispatch: Dispatch, getState?: AppState) => T
```

Comment:
`redux-thunk` in fact has typings so perhaps it'd make more sense to use `ThunkAction` from [redux-thunk/index.d.ts](https://github.com/reduxjs/redux-thunk/blob/master/index.d.ts)?

It's great you've included redux-thunk in these docs.  Definitely belongs here.